### ISSUE TITLE
Box android sdk does not seem to support metadata for folders

### ISSUE THREAD

**Issue by dshbq** [box/box-android-sdk#273]
I was checking metadata API support for BoxFolders but could not find it. I can see BoxRequestsMetadata for Files but don't see anything for Box folders. Do we have any such support for metadata for BoxFolders?
Also, i tried creating a custom class for supporting , Boxfolder , same as BoxRequestsMetadata but in values part(map), when i pass a double or string value, i start getting below error:

```
{"message":"instance type (string) does not match any allowed primitive type (allowed: [\"integer\",\"number\"]) (pointer: \/pricePaid)","code":"schema_validation_failed","request_id":"727597709590ae061e285c"}`
com.box.androidsdk.content.BoxException: An error occurred while sending the request
com.box.androidsdk.content.requests.BoxRequest.onSend(BoxRequest.java:243)
 at com.box.androidsdk.content.requests.BoxRequest.send(BoxRequest.java:190)
  at com.allstate.digitallockerpoc.activity.ItemDetailActivity$3.run(ItemDetailActivity.java:401)

```
No matter what parameter i am passing i am getting the same error, 
is Box does not support any parameter other than String or its some other kind of error?

Any help will be highly appreciated.

Thanks!!

**Comment by doncung** [comment]
Sorry for the delay. I needed to check up on this. Basically it looks like we do have metadata support for folders, but when the metadata api was added for the android sdk it was not available yet. I'm not sure about the error you are experiencing, but you should probably be able to get the folder functionality by creating copying the File methods and switching out getFilesUrl() inside BoxApiMetadata with a "getFoldersUrl" that has the same implementation with "folders" instead of "files". We'll be adding this functionality in the future.

**Comment by technicallyerik** [comment]
@doncung - The error above is because the API is validating integer template metadata values are sent over in the JSON as a numeric value, but the SDK casts numeric values to strings before sending.

**Comment by doncung** [comment]
Thanks. We'll make appropriate changes to the SDK to allow integer values.

**Comment by dshbq** [comment]
@doncung: Thanks!! Also, please make changes for other data type as well, like double, float etc. And, by what time we can expect this change to be available in the SDK?

**Comment by doncung** [comment]
I'll add it as a story to slot into our sprint planning. Depending on priorities for the team it will take about 2 weeks if we pick it up.

**Comment by dshbq** [comment]
@donucung I have forked the SDK and made changes locally for above issues and its working fine now. Please let me know how can we merge these changes to the actual SDK version

**Comment by doncung** [comment]
Thanks for your help. All you need to do is make a pull request into master. You may need to agree to our CLA if you haven't. After that we'll take a look at the pull request and add comments to it in case there are any changes that may be necessary. Otherwise we'll merge that into the code base, and it will be part of the next maven push we make.

**Comment by doncung** [comment]
I've made changes which should work. I still need to add some tests.
https://github.com/box/box-android-sdk/pull/287/files

**Comment by doncung** [comment]
This is included in 4.0.9 release.
